### ISSUE TITLE
Include CSV Dialect Description Format in csvy

### ISSUE THREAD

**Issue by CharlesNepote** [csvy/csvy.github.io#9]
Why not include [CSV Dialect Description Format](http://specs.frictionlessdata.io/csv-dialect/) in csvy?

It can be also easily expressed in YAML syntax, isn't it?

**Comment by hadley** [comment]
I think this would be a great idea - that way client libraries can eliminate all arguments to `read_csvy()` functions.

**Comment by jrovegno** [comment]
If that simplifies the specification and maintains compatibility with existing implementations, I also think it might be a good idea.
@hadley Do you have a PR of how the change would be?

**Comment by hadley** [comment]
At a minimum, I'd convert

```json
{
  "csvddfVersion": 1.0,
  "delimiter": ",",
  "doubleQuote": true,
  "lineTerminator": "\r\n",
  "quoteChar": "\"",
  "skipInitialSpace": true,
  "header": true
}
```

to yaml:

```
dialect:
  csvddfVersion: 1.0
  delimiter: ","
  doubleQuote: true
  lineTerminator: "\r\n"
  quoteChar: "\""
  skipInitialSpace: true
  header: true
```

and advertise on the home page.

That said, I'm not sure that set of options accurately captures needed variations. At a minimum I think it also needs:

* `skip`: number of lines to skip at beginning of file
* `commentChar`: if found at start of line, line is ignored
* Something to parallel `doubleQuote` for escaping with backslashes
* Something to determine if quoted `nullSequence`  should be a string or NA
* A default `encoding` to use, and statement that UTF-8 will be used if not present

And

* `delimiter` should probably be `delimiterChar` for consistency
* `lineTerminator` is rarely needed, it's almost always safe to autodetect
* `skipInitialSpace` is probably better as `stripWhiteSpace`

**Comment by jrovegno** [comment]
@CharlesNepote @hadley For me it has been a bit difficult for me to understand how the frictionlessdata specifications are composed, but in my opinion:

By definition CSVY is a simple container of a [Tabular Data Package](https://frictionlessdata.io/specs/tabular-data-package/) , Tabular Data Package builds directly on the Data Package specification.  One of the requirements in Tabular Data Package is There MUST be at least one resource, Each resource MUST be a [Tabular Data Resource](http://frictionlessdata.io/specs/tabular-data-resource/), this resource property could extends Data Resource with dialect property to describe the [CSV dialect ](http://frictionlessdata.io/specs/csv-dialect/).

In conclusion, the current definition, in my opinion, already has that property, what is difficult is how to explain how it works without so many laps.

**Comment by hadley** [comment]
I think while that be how you think of it, from the users perspective it should just be an implementation detail. I find reading the frictionless specs existing because everything is defined indirectly.
